src/vs/platform/extensions/electron-main/extensionHostStarter.ts

```diff
@@ -344,26 +459,10 @@ class UtilityExtensionHostProcess extends Disposable {
 			this._hasExited = true;
 			this._onExit.fire({ pid: this._process!.pid!, code, signal: '' });
 		}));
-		const listener = (event: electron.Event, details: electron.Details) => {
-			if (details.type !== 'Utility') {
-				return;
-			}
-			// Despite the fact that we pass the argument `seviceName`,
-			// the details have a field called `name` where this value appears
-			if (details.name === serviceName) {
-				this._logService.info(`UtilityProcess<${this.id}>: terminated unexpectedly with code ${details.exitCode}.`);
-				this._hasExited = true;
-				this._onExit.fire({ pid: this._process!.pid!, code: details.exitCode, signal: '' });
-			}
-		};
-		electron.app.on('child-process-gone', listener);
```

Comment:
This is not needed since the `child-process-gone` events for the utility process will get re-emitted as `exit` event corresponding to the process. So basically, it is enough to listen to `exit` event of `UtilityProcess` api to get notified of all terminations.

Reply:
Do we have another location where we listen on `exit` and log that with the details?

Reply:
Yup just a few lines about this one https://github.com/microsoft/vscode/blob/6a5e3aad96929a7d35e09ed8d22e87a72bd16ff6/src/vs/platform/extensions/electron-main/extensionHostStarter.ts#L342-L346